### PR TITLE
Ds shrine string fix

### PR DESCRIPTION
**Pull request by XT3RM1N8R**
Moved WaitEndSample global declaration to sound.h because it addresses sound polling and moved spell shrine handling of DoEvent case for spell shrines to HandleSpellShrine function for better reading and comprehension.

### CODE REVIEW

File: src/cpp/shared/adventure/adv.cpp [hunk 1]
```diff
@@ -122,11 +121,22 @@ void game::MakeAllWaterVisible(int player) {
 
 void advManager::DoEvent(class mapCell *cell, int locX, int locY) {
   hero *hro = &gpGame->heroes[gpCurPlayer->curHeroIdx];
-  int locType = cell->objType & 0x7F;
+  int LocationType = cell->objType & 0x7F;
```
Comment (jkoppel) [inline src/cpp/shared/adventure/adv.cpp]: The convention is that variable names should be lowercase.

File: src/cpp/shared/adventure/adv.cpp [hunk 2]
```diff
@@ -122,11 +121,22 @@ void game::MakeAllWaterVisible(int player) {
...
+  this->UpdateTownLocators(1, 1);
+  this->UpdBottomView(1, 1, 1);
+  this->UpdateScreen(0, 0);
+  gpSoundManager->SwitchAmbientMusic(giTerrainToMusicTrack[this->currentTerrain]);
+  WaitEndSample(res2, res2.sample);
+  CheckEndGame(0, 0);
+}
 
-  switch (locType) {
+void advManager::HandleSpellShrine(class mapCell *cell, int LocationType, hero *hro, SAMPLE2 res2, int locX, int locY) {
```
Comment (jkoppel) [inline src/cpp/shared/adventure/adv.cpp]: This is not what I meant at all. We can chat about this.

File: src/cpp/shared/adventure/adv.cpp (hunk 1 above)
Comment (XT3RM1N8R) [inline src/cpp/shared/adventure/adv.cpp]: I'll fix this

File: src/cpp/shared/adventure/adv.cpp (hunk 2 above)
Comment (XT3RM1N8R) [inline src/cpp/shared/adventure/adv.cpp]: I made sure to move the logic for determining whether or not a location was a shrine or not (thus, determining whether or not to handle a spell shrine event) to the main modified DoEvent function, before the potential call of HandleSpellShrine. Sorry for the confusion

**Comment by jkoppel** [comment]
You've solved the biggest issue in this code. I still have many problems with it, but it's ready for merging.

**Comment by XT3RM1N8R** [comment]
What other issues were there?

**Pull request merged**